MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -441,35 +712,58 @@ int main(int argc, char * argv[])
                                     {
+                                        errorDesc += L"Succeeded best-effort attempt to delete the incomplete VHD(X) file.";
                                         std::wcout << "Best-effort attempt to delete the incomplete VHD(X) file " << unpackDestination << " succeeded." << std::endl;
                                     }
                                 }
                             }
 
                             std::wcout << std::endl;
 
+                            // Telemetry : Workflow Log
+                            QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                            workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                            msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                             return hrCreateVHD;
                         }
 
                         // Unpack to the mounted VHD
                         std::wstring mountedUnpackDest = driveLetter + L":\\" + cli.GetRootDirectory();
-                        RETURN_IF_FAILED(MsixCoreLib::Unpack(
+                        HRESULT hrUnpackToVHD = MsixCoreLib::Unpack(
                             packageSourcePath,
                             mountedUnpackDest,
                             cli.IsApplyACLs(),
                             cli.IsValidateSignature(),
                             skippedFiles,
                             failedPackages,
                             failedPackagesErrors
-                        ));
+                        );
+
+                        if (FAILED(hrUnpackToVHD))
+                        {
+                            errorCode = msixmgr::GetErrorCodeFromHRESULT(hrUnpackToVHD);
+                            errorDesc = L"Failed unpack to the mounted vhd(x). HRESULT " + errorCode + L".";
+
+                            // Telemetry : Workflow Log
+                            QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                            workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                            msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
+                            RETURN_IF_FAILED(hrUnpackToVHD);
+                        }
 
                         HRESULT hrUnmount = MsixCoreLib::UnmountVHD(unpackDestination);
                         if (FAILED(hrUnmount))
                         {
+                            errorCode = msixmgr::GetErrorCodeFromHRESULT(hrUnmount);
+                            errorDesc = L"Successful Unpack to mounted vhd(x). But, unmounting the vhd(x) failed. HRESULT " + errorCode + L".";

```

Comment:
Same as above.

Reply:
As discussed, all the Telemetry variables will have the same Error Message as the actual Console statements. The change for outputting the Error Message on Console from a variable will be done as part of Error Message Improvement Task that will be taken later.